### ISSUE TITLE
Build failure:  python3.12-mss-9.0.2

### ISSUE THREAD

**Issue by EricWardner** [NixOS/nixpkgs#353664]
### Steps To Reproduce

1. build something with python312 mss

### Build log

```
src/tests/test_implementation.py::test_thread_safety Fatal Python error: Aborted

Current thread 0x00007ffff42fc6c0 (most recent call first):
  File "/build/mss-9.0.2/src/mss/linux.py", line 312 in __init__
  File "/build/mss-9.0.2/src/mss/factory.py", line 33 in mss
  File "/build/mss-9.0.2/src/tests/test_implementation.py", line 248 in record
  File "/nix/store/901c80rlps5q05bnjk1sj4zaz5k736nc-python3-3.12.7/lib/python3.12/threading.py", line 1012 in run
  File "/nix/store/901c80rlps5q05bnjk1sj4zaz5k736nc-python3-3.12.7/lib/python3.12/threading.py", line 1075 in _bootstrap_inner
  File "/nix/store/901c80rlps5q05bnjk1sj4zaz5k736nc-python3-3.12.7/lib/python3.12/threading.py", line 1032 in _bootstrap

Thread 0x00007ffff4afd6c0 (most recent call first):
  File "/build/mss-9.0.2/src/mss/linux.py", line 340 in close
  File "/build/mss-9.0.2/src/mss/base.py", line 57 in __exit__
  File "/build/mss-9.0.2/src/tests/test_implementation.py", line 248 in record
  File "/nix/store/901c80rlps5q05bnjk1sj4zaz5k736nc-python3-3.12.7/lib/python3.12/threading.py", line 1012 in run
  File "/nix/store/901c80rlps5q05bnjk1sj4zaz5k736nc-python3-3.12.7/lib/python3.12/threading.py", line 1075 in _bootstrap_inner
  File "/nix/store/901c80rlps5q05bnjk1sj4zaz5k736nc-python3-3.12.7/lib/python3.12/threading.py", line 1032 in _bootstrap
  
  Thread 0x00007ffff7f8f740 (most recent call first):
  File "/nix/store/901c80rlps5q05bnjk1sj4zaz5k736nc-python3-3.12.7/lib/python3.12/threading.py", line 1169 in _wait_for_tstate_lock
  File "/nix/store/901c80rlps5q05bnjk1sj4zaz5k736nc-python3-3.12.7/lib/python3.12/threading.py", line 1149 in join
  File "/build/mss-9.0.2/src/tests/test_implementation.py", line 261 in test_thread_safety
  File "/nix/store/vjbdfb8y58mmnlv8wxf6hlacfz6zih03-python3.12-pytest-8.3.3/lib/python3.12/site-packages/_pytest/python.py", line 159 in pytest_pyfunc_call
  File "/nix/store/0piy0v34vxpsw7ppzwmk04lap6c3v8wm-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/nix/store/0piy0v34vxpsw7ppzwmk04lap6c3v8wm-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/nix/store/0piy0v34vxpsw7ppzwmk04lap6c3v8wm-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/nix/store/vjbdfb8y58mmnlv8wxf6hlacfz6zih03-python3.12-pytest-8.3.3/lib/python3.12/site-packages/_pytest/python.py", line 1627 in runtest
  File "/nix/store/vjbdfb8y58mmnlv8wxf6hlacfz6zih03-python3.12-pytest-8.3.3/lib/python3.12/site-packages/_pytest/runner.py", line 174 in pytest_runtest_call
  File "/nix/store/0piy0v34vxpsw7ppzwmk04lap6c3v8wm-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/nix/store/0piy0v34vxpsw7ppzwmk04lap6c3v8wm-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/nix/store/0piy0v34vxpsw7ppzwmk04lap6c3v8wm-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/nix/store/vjbdfb8y58mmnlv8wxf6hlacfz6zih03-python3.12-pytest-8.3.3/lib/python3.12/site-packages/_pytest/runner.py", line 242 in <lambda>
  File "/nix/store/vjbdfb8y58mmnlv8wxf6hlacfz6zih03-python3.12-pytest-8.3.3/lib/python3.12/site-packages/_pytest/runner.py", line 341 in from_call
  File "/nix/store/vjbdfb8y58mmnlv8wxf6hlacfz6zih03-python3.12-pytest-8.3.3/lib/python3.12/site-packages/_pytest/runner.py", line 241 in call_and_report
  File "/nix/store/vjbdfb8y58mmnlv8wxf6hlacfz6zih03-python3.12-pytest-8.3.3/lib/python3.12/site-packages/_pytest/runner.py", line 132 in runtestprotocol
  File "/nix/store/vjbdfb8y58mmnlv8wxf6hlacfz6zih03-python3.12-pytest-8.3.3/lib/python3.12/site-packages/_pytest/runner.py", line 113 in pytest_runtest_protocol
  File "/nix/store/0piy0v34vxpsw7ppzwmk04lap6c3v8wm-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/nix/store/0piy0v34vxpsw7ppzwmk04lap6c3v8wm-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/nix/store/0piy0v34vxpsw7ppzwmk04lap6c3v8wm-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/nix/store/vjbdfb8y58mmnlv8wxf6hlacfz6zih03-python3.12-pytest-8.3.3/lib/python3.12/site-packages/_pytest/main.py", line 362 in pytest_runtestloop
  File "/nix/store/0piy0v34vxpsw7ppzwmk04lap6c3v8wm-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/nix/store/0piy0v34vxpsw7ppzwmk04lap6c3v8wm-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/nix/store/0piy0v34vxpsw7ppzwmk04lap6c3v8wm-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/nix/store/vjbdfb8y58mmnlv8wxf6hlacfz6zih03-python3.12-pytest-8.3.3/lib/python3.12/site-packages/_pytest/main.py", line 337 in _main
    File "/nix/store/vjbdfb8y58mmnlv8wxf6hlacfz6zih03-python3.12-pytest-8.3.3/lib/python3.12/site-packages/_pytest/main.py", line 283 in wrap_session
  File "/nix/store/vjbdfb8y58mmnlv8wxf6hlacfz6zih03-python3.12-pytest-8.3.3/lib/python3.12/site-packages/_pytest/main.py", line 330 in pytest_cmdline_main
  File "/nix/store/0piy0v34vxpsw7ppzwmk04lap6c3v8wm-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/nix/store/0piy0v34vxpsw7ppzwmk04lap6c3v8wm-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/nix/store/0piy0v34vxpsw7ppzwmk04lap6c3v8wm-python3.12-pluggy-1.5.0/lib/python3.12/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/nix/store/vjbdfb8y58mmnlv8wxf6hlacfz6zih03-python3.12-pytest-8.3.3/lib/python3.12/site-packages/_pytest/config/__init__.py", line 175 in main
  File "/nix/store/vjbdfb8y58mmnlv8wxf6hlacfz6zih03-python3.12-pytest-8.3.3/lib/python3.12/site-packages/_pytest/config/__init__.py", line 201 in console_main
  File "/nix/store/vjbdfb8y58mmnlv8wxf6hlacfz6zih03-python3.12-pytest-8.3.3/bin/.pytest-wrapped", line 9 in <module>
  
  Extension modules: PIL._imaging (total: 1)
/nix/store/gml76hgrj0xxz3jiqc3hk4fky4wjzlln-xvfb-run-1+g87f6705/bin/.xvfb-run-wrapped: line 181:   207 Aborted                 (core dumped) DISPLAY=:$SERVERNUM XAUTHORITY=$AUT
HFILE "$@"
```

### Additional context

Potentially related to this upsteam issue
https://github.com/BoboTiG/python-mss/issues/251

### Notify maintainers

@mweinelt

### Metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.58, NixOS, 24.11 (Vicuna), 24.11.20241025.18536bf`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.9`
 - nixpkgs: `/nix/store/wb6agba4kfsxpbnb5hzlq58vkjzvbsk6-source`
 ```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by FliegendeWurst** [comment]
Ping maintainer @austinbutler

**Comment by BoboTiG** [comment]
FYI I just fixed BoboTiG/python-mss#251 and will cut a new release.

**Comment by jmpunkt** [comment]
The fix introduced with 10.0.0 (https://github.com/NixOS/nixpkgs/pull/359644) should probably be backported to 24.11. Hydra ran into the same issue https://hydra.nixos.org/build/279678302/ . Then I had the same error on my machine.
